src/fsi/console.fs

```diff
@@ -104,6 +104,74 @@ module internal Utils =
             | ' ', false -> nextWordFromIdx line (idx + 1, false)
             | _, _ -> nextWordFromIdx line (idx + 1, true)
 
+    /// An array stores ranges of full-width chars.
+    ///
+    /// Array [| [| a; b |]; [| c; d |] |] represents range [a, b] or [c, d], means chars in these ranges are full-width.
```

Comment:
It's actually just `[| a; b; c; d |]`. Don't know what's less confusing :)

Reply:
This may be more clear. 

``` fsharp
type CharRange = { From: char; To: char }

/// An array stores ranges of full-width chars.
let private fullWidthCharRanges =
    seq {
        { From = '\u1100'; To = '\u115f' }
        { From = '\u231a'; To = '\u231b' }
        { From = '\u2329'; To = '\u232a' }
        ...
    }
    |> Seq.fold
        (fun (state: ResizeArray<_>) { From = i; To = j } ->
            state.Add(i)
            state.Add(j)
            state)
        (ResizeArray())
```

Reply:
In order to use existing `BinarySearch` function to determine whether a char is in the ranges, the ranges need to be converted to `Array` or `ResizeArray`

Reply:
I think your version with the comments is fine, just to make the comments accurate.

Reply:
How about this

``` fsharp
/// An array stores ranges of full-width chars.
/// 
/// The ranges are sorted by increasing order in the array, and each range are stored in the 2nth and 2n+1th 
/// position in the array (n is the ordinal number of the range)
/// 
/// Array [| a; b; c; d |] represents range [a, b] or [c, d], means chars in these ranges are full-width.
let private fullWidthCharRanges =
```

Reply:
If possible, it would be good to explain in the comment where these ranges come from, e.g. point to the relevant part of unicode spec or some other authorative source